engines/scc_proxy/lib/scc_proxy/engine.rb

```diff
@@ -429,6 +432,15 @@ def handle_response(response)
             raise ActionController::TranslatedError.new(error['error'])
           end
         end
+
+        def system_has_paid_extension?(scc_systems_activations)
+          active_products_classes = scc_systems_activations.map do |act|
+            if act['status'].casecmp('active').zero? && act['service']['product']['product_class'] != @product.product_class
+              act['service']['product']['product_class']
+            end
+          end.flatten.compact
+          active_products_classes.present?
+        end
```

Comment:
I don't think there's a need to check product classes here: 

```suggestion
        def system_has_paid_extension?(scc_systems_activations)
          # drop the just de-activated activation from the list to avoid another call to SCC
          scc_systems_activations.reject!{|act| act['service']['product']['id'] == @product.id }
          scc_systems_activations.present?
        end
```